edge-util/src/Microsoft.Azure.Devices.Edge.Storage/StoreProvider.cs

```diff
@@ -25,10 +25,22 @@ public StoreProvider(IDbStoreProvider dbStoreProvider, TimeSpan operationTimeout
         public IEntityStore<TK, TV> GetEntityStore<TK, TV>(string entityName)
         {
             IDbStore entityDbStore = this.dbStoreProvider.GetDbStore(Preconditions.CheckNonWhiteSpace(entityName, nameof(entityName)));
-            IKeyValueStore<TK, TV> dbStoreMapper = new KeyValueStoreMapper<TK, byte[], TV, byte[]>(entityDbStore, new BytesMapper<TK>(), new BytesMapper<TV>());
-            IEntityStore<TK, TV> entityStore = new EntityStore<TK, TV>(dbStoreMapper, entityName, 12);
-            IEntityStore<TK, TV> timedEntityStore = new TimedEntityStore<TK, TV>(entityStore, this.operationTimeout);
-            return timedEntityStore;
+            return this.GetEntityStore<TK, TV>(entityName, entityDbStore);
+        }
+
+        public IEntityStore<TK, TV> GetEntityStore<TK, TV>(string backwardCompatibleEntityName, string entityName)
+        {
+            Preconditions.CheckNonWhiteSpace(backwardCompatibleEntityName, nameof(backwardCompatibleEntityName));
```

Comment:
Do we need a test for this logic?

Reply:
The tests from EntityStoreTestBase cover this, I added: BackwardCompatibleStoreName_ShouldReturnOlderEntityStore and BackwardCompatibleStoreName_ShouldReturnNewEntity